pkg/ddl/db_table_test.go

```diff
@@ -344,6 +344,64 @@ func TestCreateTableWithInfo(t *testing.T) {
 	require.Greater(t, idGenNum, id)
 }
 
+func TestCreateTableWithEngineAttribute(t *testing.T) {
+	testfailpoint.Enable(t, "github.com/pingcap/tidb/pkg/infoschema/mockTiFlashStoreCount", `return(true)`)
+	defer testfailpoint.Disable(t, "github.com/pingcap/tidb/pkg/infoschema/mockTiFlashStoreCount")
+
+	store := testkit.CreateMockStoreWithSchemaLease(t, 200*time.Millisecond)
+	tk := testkit.NewTestKit(t, store)
+	rpcserver, _ := setUpRPCService(t, "127.0.0.1:0", domain.GetDomain(tk.Session()), nil)
+	defer rpcserver.Stop()
+
+	tk.MustExec("use test;")
+	tk.MustExecToErr("CREATE TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ENGINE_ATTRIBUTE = '{\"key\": \"value}';")                                              // invalid json
+	tk.MustExecToErr("CREATE TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ENGINE_ATTRIBUTE = '{\"key\": \"value\"}';")                                            // invalid key
+	tk.MustExecToErr("CREATE TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ENGINE_ATTRIBUTE = '{\"tiflash-replica\": \"2\"}';")                                    // invalid value
+	tk.MustExecToErr("CREATE TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ENGINE_ATTRIBUTE = '{\"tiflash-replica\": 13}';")                                       // invalid value
+	tk.MustExecToErr("CREATE TEMPORARY TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ENGINE_ATTRIBUTE = '{\"tiflash-replica\": 2}';")                              // not support temporary table
+	tk.MustExecToErr("CREATE GLOBAL TEMPORARY TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ON COMMIT DELETE ROWS ENGINE_ATTRIBUTE = '{\"tiflash-replica\": 2}';") // not support temporary table
+
+	tk.MustExec("CREATE TABLE t (id INT PRIMARY KEY, value VARCHAR(16383)) ENGINE_ATTRIBUTE = '{\"tiflash-replica\": 2}';") // valid
+	tk.MustQuery("SELECT REPLICA_COUNT from information_schema.tiflash_replica where table_schema='test' and table_name='t'").Check(testkit.Rows("2"))
+	tk.MustQuery("SHOW CREATE TABLE t").Check(testkit.Rows(
+		"t CREATE TABLE `t` (\n" +
+			"  `id` int(11) NOT NULL,\n" +
+			"  `value` varchar(16383) DEFAULT NULL,\n" +
+			"  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */\n" +
+			") ENGINE=InnoDBENGINE_ATTRIBUTE=`{\"columnar-replica\": 2}` DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
```

Comment:
```suggestion
			") ENGINE=InnoDB ENGINE_ATTRIBUTE=`{\"columnar-replica\": 2}` DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
```
@Lloyd-Pottiger There should be a space between "ENGINE=InnoDB" and "ENGINE_ATTRIBUTE=..."